pkg/service/service.go

```diff
@@ -2135,3 +2151,26 @@ func (s *Service) SyncNodePortFrontends(addrs sets.Set[netip.Addr]) error {
 	}
 	return nil
 }
+
+func backendToNode(b *lb.Backend) *nodeTypes.Node {
+	return &nodeTypes.Node{
+		Name: fmt.Sprintf("backend-%s", b.L3n4Addr.AddrCluster.AsNetIP()),
+		IPAddresses: []nodeTypes.Address{{
+			Type: addressing.NodeInternalIP,
+			IP:   b.L3n4Addr.AddrCluster.AsNetIP(),
+		}},
+	}
+}
```

Comment:
This is unnecessary conversion. AFAICS, there is nothing special included in the `node` structure. How about refactoring the common code in a separate package that handles {insert,delete}Neighbor of managed neighbor entries? The code can be reused by both `pkg/service` and `pkg/linux/node`?

Reply:
I can give it a try to refactor the whole neighbor package code and then also propagate errors, this is however a bigger lift. Can this be done independently of this work or does it need to block?

Reply:
Ok, it'll be good to have it as a follow-up.